### PR TITLE
Refactor rollCallValidator

### PR DESCRIPTION
**Pull request by Ajkunas**
Here is a first version of refactoring the validator of RollCAll. As suggested, I tried to implement a list of checks instead of having a cascade of `if else`: 
- I implemented functions such as `checkId` in `MessageDataContentValidator.scala` file. This type of function returns a GraphMessage, so I give them as parameter the `rpcMessage` and the error message. 
- If preferred, we can reuse the functions such as `validateTimestampOrder` instead of implemented a new function (i.e. `checkTimestampOrder`). It just needs some modifications. (I added new functions so that it does not impact the other validator files). 
- I modified slightly the private functions `validateOpens` and `validateCloses` so that it returns a GraphMessage and so that I can use them in the list. 
- The function `runList` runs the checks through the entire list. It stops when we get an error (i.e the check returns a `Right(PipelineError(...))`), and so the validator returns this error. If the function goes through the list without any error, it simply returns `Left(rpcMessage)`. 
- I did not add any comments, but, of course, it should be done. 

I am open to any suggestion.

### CODE REVIEW

**Review by pierluca — COMMENTED** [review]
In general, this seems much better and much more manageable.

However, I'm concerned by the length of each code line.
It makes things hard to read and hard to review.
Can we stick to the 80 characters limit that's defined in the .editorconfig ?
Could it be that your editor isn't picking that up? 🤔 

For the `if / Left / else / Right` structures, I'm also wondering if a few newlines wouldn't massively increase readability 😁

**Comment by pierluca** [comment]
My bad, .editorconfig says 120 chars ... but that's still not respected, I believe😅

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MessageValidator.scala [hunk 1]
```diff
@@ -80,4 +88,25 @@ object MessageValidator extends ContentValidator with AskPatternConstants {
...
+
+    (data, laoId, sender, channel)
+  }
+
+  def runList(list: List[GraphMessage]): GraphMessage = {
+    if (list.head.isLeft && !list.tail.isEmpty)
+      runList(list.tail)
+    else
+      list.head
+  }
```
[the remark this answers is not in the document]
Comment (Ajkunas) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MessageValidator.scala:111]: Indeed, it seems a more appropriate name.

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MessageValidator.scala [hunk 2]
```diff
@@ -14,6 +14,37 @@ import scala.util.Success
...
+
+  /** Runs the multiple checks of the validators
+    *
+    * @param list
+    *   List of checks which return a GraphMessage
+    * @return
+    *   GraphMessage: passes the rpcMessages to Left if successful right with pipeline error
+    */
+  def runChecks(list: List[GraphMessage]): GraphMessage = {
+    if (list.head.isLeft && !list.tail.isEmpty)
```
Comment (Ajkunas) [inline be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MessageValidator.scala:42]: I moved these functions at the top of the file. In my opinion, it seems more clear to have them before all the checks. 

I also added comments here and there, and changed `runList` to `runChecks` as suggested.

File: be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MessageValidator.scala [hunk 3]
```diff
@@ -80,4 +88,25 @@ object MessageValidator extends ContentValidator with AskPatternConstants {
...
+    val message: Message = rpcMessage.getParamsMessage.get
+    val data: T = message.decodedData.get.asInstanceOf[T]
+    val laoId: Hash = rpcMessage.extractLaoId
+    val sender: PublicKey = message.sender
+    val channel: Channel = rpcMessage.getParamsChannel
+
+    (data, laoId, sender, channel)
+  }
+
+  def runList(list: List[GraphMessage]): GraphMessage = {
```
[the remark this answers is not in the document]
Comment (Ajkunas) [reply be2-scala/src/main/scala/ch/epfl/pop/pubsub/graph/validators/MessageValidator.scala]: I did some research about it and found a way to do as you suggested. Clearly, we avoid the `List(...)` which is not really useful.

**Review by pierluca — APPROVED** [review]
This feels much much cleaner, great job !

**Pull request merged**